### ISSUE TITLE
Stream quality setting adjustment based on dropped frames count

### ISSUE THREAD

**Issue by mjarvisal** [peak3d/inputstream.adaptive#164]
I've been spending today getting Netflix running on my Raspberry Pi 3 B with OSMC and while being able to select the stream quality manually is nice it would be even nicer if the stream quality would also change depending on if a resource constrained system like the PI can actually decode the stream in the original fps.

So my improvement suggestion would be that the stream quality would change depending on achieved fps and free resources as well as bandwidth.

**Comment by peak3d** [comment]
Yes, correct stream selection is still a field which has to be improoved.
First /easy step would be for example looking at the whitelist and only select resolutions which meet the whitelist restictions (settings::system::video::whitelist), but it could be also tracked on runtime.

Unfortunately there have been too many issues regarding "why does my stream does not play" so no time was left for this part.

**Comment by matthuisman** [comment]
@peak3d

Would it also be possible for addons / plugins to tell Inputstream what minimum or maximum quality to use?

So, say a user selected they wanted Max: 720p in my addons settings.
I could tell Inputstream to use anything up-to 720.

```
#KODIPROP:inputstream.adaptive.max_quality=720p
li.setProperty('inputstream.adaptive.max_quality', '720p')
```

Or something like that?

**Comment by CastagnaIT** [comment]
there are many possibilities that could be implemented as rules for the automatic selection of stream quality, as fps, available memory, cpu load percentage, cdn selection etc...

this is require a complex development, one of major problems is that we do not have direct access to that information and so require also changes on Kodi core in the hope that they will be accepted when proposed

in the near future this cannot be implemented and we cannot predict when this will be possible do it, then i close this issue,
there is no point in leaving an issue pending for years
